drawer: Update Do and Don't content (#1562)
agriculturegovau/agds-next: packages/react/src/drawer/docs/overview.mdx

<DoHeading />

- consider using a side action page first - side action pages support more use cases than side drawers

> - only use side drawers for filters, small overflow form actions and related guidance content if you can’t provide these interactions on the parent page

Author: ```suggestion
- only use side drawers for filters, small overflow form actions and related guidance content
```

> - only use side drawers for external facing forms when you need to maintain the user’s context of the parent page

Author: ```suggestion
- only use for external (customer-facing) forms if there is a need to maintain some context of the current page, otherwise link to a separate page
```

- read the [Search filters pattern](/patterns/search-filters)
- use a [Button group](/components/button#button-group) to display actions
- include a meaningful title - e.g. ‘Filter by’
- optionally close the Drawer when an action button is pressed

<DontHeading />

> - use for long forms or any content that is likely to cause a user to scroll vertically or horizontally in the drawer - use a side action page for these interactions instead

Author: ```suggestion
- use for long forms or any content that is likely to cause a user to scroll vertically or horizontally in the drawer - link to a separate page instead
```

> - remove the ‘Close’ button

Author: ```suggestion
- provide navigation or multiple pages inside a drawer
- remove the ‘Close’ button
```

Reviewer: "provide navigation" is a bit vague. Because we do allow put links to content pages afaik (like in the help context feature Pete worked on)

<DoHeading />

> - consider using a side action page first - side action pages support more use cases than side drawers

Author: ```suggestion
- consider using a side action page first - side action pages support more use cases than side drawers
```
- consider linking to another page before using a side drawer

> - consider using a side action page first - side action pages support more use cases than side drawers

Reviewer: ```suggestion
- consider using a side action page first – side action pages support more use cases than side drawers
```

- only use side drawers for filters, small overflow form actions and related guidance content if you can't provide these interactions on the parent page

> - only use side drawers for external facing forms when you need to maintain the users context of the parent page 

Author: ```suggestion
- only use side drawers for external facing forms when you need to maintain the users context of the parent page 
```
-only use for external (customer-facing) forms if there is a need to maintain some context of the current page, otherwise link to a separate page
